collections/_article/dev-snapshot-godot-4-2-beta-1.md

```diff
@@ -0,0 +1,382 @@
+
+There are still some kinks to iron out, some [caveats to consider](https://docs.godotengine.org/en/latest/tutorials/scripting/c_sharp/index.html#c-platform-support). And of course we rely on public testing to make sure everything works as expected. Until 4.3 this support will remain experimental, and we are still missing the web platform. But we are confident that the current state is good enough to plan ahead if your project is expected to ship to iOS or Android, so please start testing and report what you find!
+
+In more general improvements to the C# integration, there are a number of fixes to the binding generator by [Thaddeus Crews](https://github.com/Repiteo) ([GH-80628](https://github.com/godotengine/godot/pull/80628), [GH-80630](https://github.com/godotengine/godot/pull/80630), [GH-80631](https://github.com/godotengine/godot/pull/80631)). Bindings also support abstract classes now ([GH-81101](https://github.com/godotengine/godot/pull/81101) by [398utubzyt](https://github.com/398utubzyt)). Interoperability with GDScript has improved thanks to additions by [Zae Chao](https://github.com/zaevi) and [wscalf](https://github.com/wscalf), who contributed support for static methods and readonly/writeonly properties ([GH-81783](https://github.com/godotengine/godot/pull/81783), [GH-67304](https://github.com/godotengine/godot/pull/67304)).
+
+[Raul Santos](https://github.com/raulsntos) put some time into updating the looks of the build panel in the editor ([GH-80260](https://github.com/godotengine/godot/pull/80260)), and bringing it to functional parity with the main output log and debugger. He also leveraged GDExtension's compatibility API to ensure C# bindings also include the same compatibility methods with little to no intervention from other contributors ([GH-80527](https://github.com/godotengine/godot/pull/80527)).
+
+#### GDScript
+
+GDScript offers the best of both worlds when it comes to typing your variables. You can go completely dynamic, but you can also gradually introduce more and more strict types to your codebase. This is paramount when it comes to iteration speed and turning your prototypes into fleshed-out projects.
+
+This release comes with a few additions to empower your statically typed code. Primary among them is the new warning that reports all cases of untyped code so you can make sure your codebase is strict and consistent ([GH-81355](https://github.com/godotengine/godot/pull/81355) by [Ryan Brue](https://github.com/ryanabx)). To help you ensure you never trigger this warning, `for` loops now support static typing ([GH-80247](https://github.com/godotengine/godot/pull/80247)), and you can use `preload`ed scripts as type hints as well ([GH-80964](https://github.com/godotengine/godot/pull/80964)). [Danil Alexeev](https://github.com/dalexeev) has been killing it with fixes and new language features lately!
```

Comment:
Member constants supported this before, the change is that now local constants support this too. Not only a `preload`ed script is considered a type, but also any other type (class or enum).